src/main.h

```diff
@@ -128,9 +128,9 @@ inline unsigned int GetTargetSpacing(int nHeight) { return IsProtocolV2(nHeight)
 extern bool IsNeuralNodeParticipant(const std::string& addr, int64_t locktime);
 bool VerifySuperblock(const std::string& superblock, const CBlockIndex* parent);
 
-extern std::map<std::string, double> mvNeuralNetworkHash;
-extern std::map<std::string, double> mvCurrentNeuralNetworkHash;
-extern std::map<std::string, double> mvNeuralVersion;
+extern std::unordered_map<std::string, double> mvNeuralNetworkHash;
```

Comment:
imo we should `typedef std::unordered_map<std::string, double> NeuralMap`

would make this a little neater and then we can do the `NeuralMap::iterator`

Think this would look cleaner. 

We should consider the same in another PR for `std::map<std::string, StructCPID>` if we can as well.

liking the use of unordered_map for optimizations

Reply:
I'll do a sweep of all the maps in a later PR.